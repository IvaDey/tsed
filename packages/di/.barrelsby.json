{
  "directory": ["./src/common", "./src/node"],
  "exclude": ["**/__mock__", "**/__mocks__", "**/*.spec.ts", "localsContainer.ts"],
  "delete": true
}
