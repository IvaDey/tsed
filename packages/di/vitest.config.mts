// @ts-ignore
import {presets} from "@tsed/vitest/presets";
import {defineConfig} from "vitest/config";

export default defineConfig(
  {
    ...presets,
    test: {
      ...presets.test,
      coverage: {
        ...presets.test.coverage,
        thresholds: {
          statements: 98.69,
          branches: 97.18,
          functions: 97.02,
          lines: 98.69
        }
      }
    }
  }
);
