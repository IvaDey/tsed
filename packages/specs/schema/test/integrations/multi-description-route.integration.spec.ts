import "../../src/index.js";

import {Controller} from "@tsed/di";

import {getSpec, Name, Put, SpecTypes} from "../../src/index.js";

@Controller("/")
class TestController {
  @(Put("/").Description("Update a pet"))
  @(Put("/:id").Description("Description with id"))
  @(Put("/:id/:albumToken").Description("Description with id and albumToken"))
  put() {
    return null;
  }
}

describe("Multi description", () => {
  describe("OpenSpec", () => {
    it("should generate the spec", () => {
      const spec = getSpec(TestController, {specType: SpecTypes.OPENAPI});

      expect(spec).toMatchSnapshot();
    });
  });
});
