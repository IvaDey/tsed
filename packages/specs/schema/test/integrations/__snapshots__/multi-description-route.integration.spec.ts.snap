// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Multi description > OpenSpec > should generate the spec 1`] = `
{
  "paths": {
    "/": {
      "put": {
        "description": "Update a pet",
        "operationId": "testControllerPut_1",
        "parameters": [],
        "responses": {
          "200": {
            "description": "Success",
          },
        },
        "tags": [
          "TestController",
        ],
      },
    },
    "/{id}": {
      "put": {
        "description": "Description with id",
        "operationId": "testControllerPutById",
        "parameters": [
          {
            "in": "path",
            "name": "id",
            "required": true,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "description": "Success",
          },
        },
        "tags": [
          "TestController",
        ],
      },
    },
    "/{id}/{albumToken}": {
      "put": {
        "description": "Description with id and albumToken",
        "operationId": "testControllerPut",
        "parameters": [
          {
            "in": "path",
            "name": "id",
            "required": true,
            "schema": {
              "type": "string",
            },
          },
          {
            "in": "path",
            "name": "albumToken",
            "required": true,
            "schema": {
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "description": "Success",
          },
        },
        "tags": [
          "TestController",
        ],
      },
    },
  },
  "tags": [
    {
      "name": "TestController",
    },
  ],
}
`;
